include/cppkafka/kafka_handle_base.h

```diff
@@ -262,6 +262,34 @@ class CPPKAFKA_API KafkaHandleBase {
      */
     int get_out_queue_length() const;
 
+#if RD_KAFKA_VERSION >= RD_KAFKA_DESTROY_FLAGS_SUPPORT_VERSION
+    /**
+     * \brief Sets destroy_flags
+     *
+     * Destroy flags will be used for handle destruction.
+     *
+     * Single values: 0 (Default), RD_KAFKA_DESTROY_F_NO_CONSUMER_CLOSE
+     *
+     * Setting to RD_KAFKA_DESTROY_F_NO_CONSUMER_CLOSE prevents potential
+     * hang during termination if consumer.close() is failing because
+     * of queued callback calls.
+     *  Refs:
+     * https://github.com/edenhill/librdkafka/blob/7269f0c6a225cd42dc7a44c5e081c464986d20ff/tests/0084-destroy_flags.c#L69
+     * https://github.com/edenhill/librdkafka/issues/2077
+     */
+     void set_destroy_flags(int destroy_flags) {
+         destroy_flags_ = destroy_flags;
+     };
+
+    /**
+     * \brief Returns destroy_flags
+     *
+     */
+     int get_destroy_flags() const {
+         return destroy_flags_;
```

Comment:
AFAICS implementation part even for oneline functions are in module part in cppkafka